FlowCal/plot.py

```diff
@@ -1061,6 +1302,1271 @@ def hist1d(data_list,
+            positions = 10**positions
+        positions_length = len(positions)
+    else:
+        try:
+            positions_length = len(positions)
+        except TypeError:
+            positions = [positions]
+            positions_length = 1
+
+    if positions_length != data_length:
+        msg  = "`positions` must have the same length as `data`"
+        raise ValueError(msg)
+
+    # separately illustrate position=0 if log x-axis
+    if xscale == 'log' and 0 in list(positions):
+        data      = list(data)
+        positions = list(positions)
+
+        zero_idx = [idx
+                    for idx,pos in enumerate(positions)
+                    if pos == 0]
+
+        if len(zero_idx) > 1:
+            msg  = "attempting to separately illustrate position=0 violin,"
+            msg += " but found multiple instances"
+            raise ValueError(msg)
+        zero_idx = zero_idx[0]
+
+        zero_data = data.pop(zero_idx)
+        del positions[zero_idx]
+        data_length      = len(data)
+        positions_length = len(positions)
+
+        if logx_zero_data is None:
+            logx_zero_data = zero_data
+
+        # convert parameters specified via sequences to mappings
+        if isinstance(violin_kwargs, collectionsAbc.Sequence):
+            violin_kwargs_seq = list(violin_kwargs)
+            zero_kwargs = violin_kwargs_seq.pop(zero_idx)
+            violin_kwargs = {pos:kwargs
+                             for pos,kwargs in zip(positions,
+                                                   violin_kwargs_seq)}
+            violin_kwargs['logx_zero'] = zero_kwargs
+
+        if isinstance(draw_summary_stat_kwargs, collectionsAbc.Sequence):
+            draw_summary_stat_kwargs_seq = list(draw_summary_stat_kwargs)
+            zero_kwargs = draw_summary_stat_kwargs_seq.pop(zero_idx)
+            draw_summary_stat_kwargs = \
+                {pos:kwargs
+                 for pos,kwargs in zip(positions,
+                                       draw_summary_stat_kwargs_seq)}
+            draw_summary_stat_kwargs['logx_zero'] = zero_kwargs
+
+        if y_bin_edges is not None:
+            try:
+                first_element = next(iter(y_bin_edges))
+                try:
+                    iter(first_element)   # success => sequence of sequences
+
+                    y_bin_edges_seq = list(y_bin_edges)
+                    zero_y_bin_edges = y_bin_edges_seq.pop(zero_idx)
+                    y_bin_edges = {pos:ybe
+                                   for pos,ybe in zip(positions,
+                                                      y_bin_edges_seq)}
+                    y_bin_edges['logx_zero'] = zero_y_bin_edges
+                except TypeError:
+                    # sequence of scalars
+                    pass
+            except TypeError:
+                msg  = "`y_bin_edges` should be iterable sequence or sequence"
+                msg += " of sequences"
+                raise TypeError(msg)
```

Comment:
Sounds like this should have been checked outside the `if xscale == 'log' and 0 in list(positions):` first.

Reply:
`y_bin_edges` is understood on a per-violin basis, and the behavior is slightly different for `data` violins versus `min`, `max`, and `logx_zero` violins. So there is no global check of `y_bin_edges` like with some of the other parameters above. The only reason it's checked here is to cover the edge case where the user specified a sequence of individual `y_bin_edges` for each violin and we need to extract the `y_bin_edges` for the zero violin (the surrounding code does something similar for the other parameters).